cpp/src/io/parquet/page_data.cu

```diff
@@ -250,10 +251,11 @@ CUDF_KERNEL void __launch_bounds__(decode_block_size)
 
   page_state_s* const s = &state_g;
   auto* const sb        = &state_buffers;
-  int page_idx          = blockIdx.x;
-  int t                 = threadIdx.x;
-  int out_thread0;
-  [[maybe_unused]] null_count_back_copier _{s, t};
+  int const page_idx    = cg::this_grid().block_rank();
+  auto const block      = cg::this_thread_block();
+  auto const warp       = cg::tiled_partition<cudf::detail::warp_size>(block);
+  int out_warp_id;
```

Comment:
@vuule updated this previously missed function. This one has a bit more complex logic than others.